src/common/types.cpp

```diff
@@ -892,16 +945,234 @@ idx_t StructType::GetChildCount(const LogicalType &type) {
 	return StructType::GetChildTypes(type).size();
 }
 
-LogicalType LogicalType::STRUCT(child_list_t<LogicalType> children) {
+LogicalType LogicalType::STRUCT(const child_list_t<LogicalType> &children) {
```

Comment:
This shouldn't be a const reference since this function takes ownership of the object: the children are `moved` into the `StructTypeInfo`